### PR TITLE
Updating Torch version for kfto cuda image

### PR DESCRIPTION
**Pull request by Fiona-Waters**
## Description

The torch version provided in the previous pipefile is a pre-release or beta version which cannot be pinned correctly and updated in Pipfile.lock.

## How Has This Been Tested?

I have confirmed with @oksanabaza that this is compatible with other dependencies used for this image.
## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: images/runtime/training/cuda/Pipfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ datasets = ">=2.15.0"
 transformers = ">=4.41.2"
 numpy = "<2.0.0,>=1.23.5"
 accelerate = ">=0.34.2"
-torch = "==2.3.0a0"
+torch = "==2.4.1"
```
Comment (astefanutti) [inline images/runtime/training/cuda/Pipfile:12]: The label(s) have to be updated accordingly.

Comment (astefanutti) [reply images/runtime/training/cuda/Pipfile:12]: And the README as well :)

Comment (Fiona-Waters) [reply images/runtime/training/cuda/Pipfile:12]: Apologies. I am trying to do too many things at once. Updated labels and README. Thank you

Comment (astefanutti) [reply images/runtime/training/cuda/Pipfile:12]: No worries, I know what this is and this is what reviews are for 😃.

**Review by astefanutti — APPROVED** [review]

**Pull request merged**